mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -110,6 +149,9 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
   using OpRewritePattern<T>::OpRewritePattern;
 
   LogicalResult matchAndRewrite(T op, PatternRewriter &b) const override {
+    if (miopen::ConvOpType::Conv2DBwdDataOpType == convOpType) {
+      return backwardData(op, b);
```

Comment:
As the number of lines for `miopen.conv2d_bwd_data` is pretty large, please consider move it to another header file, and use specialization instead of testing `ConvOpType` here in the general template.